MatrixSDK/MXSession.m

```diff
@@ -2802,9 +2793,22 @@ - (MXRoom *)getOrCreateRoom:(NSString *)roomId notify:(BOOL)notify
     {
         room = [self createRoom:roomId notify:notify];
     }
+    //  also create the room summary if needed
+    [self getOrCreateRoomSummary:roomId];
```

Comment:
No we don't have to but i'd prefer it this way. Please note that this method is only called at initial sync and in case of a new room introduced. We don't load all summaries on every startup, it's managed by `-[MXSession loadRoom:]` method.

Another drawback of creating a summary when it's needed would be creating a room summary for any room id. So in that case, if we call `[mxSession roomSummaryWithRoomId:@"anything"]` we'd get a summary object even for _unknown_ rooms, which i'd like to avoid too. Right now the implementation first looks in the cache, secondly looks in the store, if not found in neither one simply returns nil.

Reply:
ok. Thank you